examples/shell-override-versions.nix

```diff
@@ -0,0 +1,25 @@
+# A standalone shell definition that overrides the versions of ESP-IDF and the ESP32 toolchain.
+
+let
+  nixpkgs-esp-dev = builtins.fetchGit {
+    url = "https://github.com/mirrexagon/nixpkgs-esp-dev.git";
+  };
+
+  pkgs = import <nixpkgs> { overlays = [ (import "${nixpkgs-esp-dev}/overlay.nix") ]; };
+in
+pkgs.mkShell
+{
+  name = "esp-project";
+
+  buildInputs = with pkgs; [
+    (gcc-xtensa-esp32-elf-bin.override {
+      version = "2021r2";
+      hash = pkgs.lib.fakeHash;
```

Comment:
This hash and the one below will be updated after merging the pull request, so I can use the shell directly - it assumes that master has this PR merged.